src/Data/LLVM/BitCode/BitString.hs

```diff
@@ -101,15 +128,17 @@ fromBitString (BitString l i) =
   case bitSizeMaybe x of
     Nothing -> x
     Just n
-      | 0 <= ival && ival < bit n -> x
+      -- Verify that the bitstring size is less than the target size, or if it is
+      -- greater, that the extra upper bits are all zero.
+      | n >= bitCount l || (ival < bit n) -> x
       | otherwise -> error (unwords
            [ "Data.LLVM.BitCode.BitString.fromBitString: bitstring value of length", show l
-           , "(", show i, ")"
+           , "(mask=0x" <> showHex i ")"
            , "could not be parsed into type with only", show n, "bits"
            ])
  where
- x    = fromInteger ival
- ival = i .&. maskBits l
+ x    = fromInteger ival  -- use Num's conversion
```

Comment:
It's not necessarily to an `Int`.  This function is polymorphic in the return type, as long as the return type conforms to `Num`, so it's round-tripping through `Integer` to get to the return type, as mentioned in the haddocks.   I added some additional comments, let me know if you think it could still use clarification.

Reply:
Oops, I missed the fact that this was polymorphic. I like the new comments.